app/controllers/api/messaging.php

```diff
@@ -0,0 +1,2601 @@
+        }
+
+        $topic = $dbForProject->updateDocument('topics', $topicId, $topic);
+
+        $queueForEvents
+            ->setParam('topicId', $topic->getId());
+
+        $response
+            ->dynamic($topic, Response::MODEL_TOPIC);
+    });
+
+App::delete('/v1/messaging/topics/:topicId')
+    ->desc('Delete a topic.')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'topic.delete')
+    ->label('audits.resource', 'topic/{request.$topicId}')
+    ->label('event', 'topics.[topicId].delete')
+    ->label('scope', 'topics.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'deleteTopic')
+    ->label('sdk.description', '/docs/references/messaging/delete-topic.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_NOCONTENT)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_NONE)
+    ->param('topicId', '', new UID(), 'Topic ID.')
+    ->inject('queueForEvents')
+    ->inject('dbForProject')
+    ->inject('queueForDeletes')
+    ->inject('response')
+    ->action(function (string $topicId, Event $queueForEvents, Database $dbForProject, Delete $queueForDeletes, Response $response) {
+        $topic = $dbForProject->getDocument('topics', $topicId);
+
+        if ($topic->isEmpty()) {
+            throw new Exception(Exception::TOPIC_NOT_FOUND);
+        }
+
+        $dbForProject->deleteDocument('topics', $topicId);
+
+        $queueForDeletes
+            ->setType(DELETE_TYPE_TOPIC)
+            ->setDocument($topic);
+
+        $queueForEvents
+            ->setParam('topicId', $topic->getId());
+
+        $response
+            ->setStatusCode(Response::STATUS_CODE_NOCONTENT)
+            ->noContent();
+    });
+
+App::post('/v1/messaging/topics/:topicId/subscribers')
+    ->desc('Create a subscriber.')
+    ->groups(['api', 'messaging'])
+    ->label('audits.event', 'subscriber.create')
+    ->label('audits.resource', 'subscriber/{response.$id}')
+    ->label('event', 'topics.[topicId].subscribers.[subscriberId].create')
+    ->label('scope', 'subscribers.write')
+    ->label('sdk.auth', [APP_AUTH_TYPE_JWT, APP_AUTH_TYPE_SESSION, APP_AUTH_TYPE_ADMIN, APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'messaging')
+    ->label('sdk.method', 'createSubscriber')
+    ->label('sdk.description', '/docs/references/messaging/create-subscriber.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_CREATED)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_SUBSCRIBER)
+    ->param('subscriberId', '', new CustomId(), 'Subscriber ID. Choose a custom Subscriber ID or a new Subscriber ID.')
+    ->param('topicId', '', new UID(), 'Topic ID. The topic ID to subscribe to.')
+    ->param('targetId', '', new UID(), 'Target ID. The target ID to link to the specified Topic ID.')
```

Comment:
@fanatic75 This param doesn't make sense for the client side, are we able to auto-detect this in that case?

Reply:
Targets can be iterated using getAccount api as we have discussed, closing this one